Storage/Storage/Bookmarks.swift

```diff
@@ -34,14 +36,11 @@ public struct BookmarkRoots {
 /**
  * The immutable base interface for bookmarks and folders.
  */
-@objc public protocol BookmarkNode {
-    var guid: String { get }
-    var title: String { get }
-    var icon: UIImage { get }
-}
-
-@objc public protocol Bookmark : BookmarkNode {
-    var url: String! { get }
+public class BookmarkNode {
+    public var id: Int? = nil
+    public var guid: String = ""
```

Comment:
Defaulting to an empty string seems like a bad idea here. Am I correct in assuming that all nodes will have a GUID? If so, make this implicitly unwrapped so we don't mask errors in our code. If not, make it optional.